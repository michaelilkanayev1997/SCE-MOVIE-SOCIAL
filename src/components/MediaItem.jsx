import PlayArrowIcon from "@mui/icons-material/PlayArrow";
import { useEffect, useState } from "react";
import { Link } from "react-router-dom";
import Button from "./Button";
import photoComingSoon from "../assets/photo-coming-soon.jpg";
import apiConfig from "../api/apiConfig";
import { UserAuth } from "../context/AuthContext";
import Typography from "@mui/material/Typography";
import ReadMoreIcon from "@mui/icons-material/ReadMore";
import { OutlineButton } from "../components/Button";

const MediaItem = ({ media, mediaType }) => {
  const [title, setTitle] = useState("");
  const [posterPath, setPosterPath] = useState("");
  const [releaseDate, setReleaseDate] = useState(null);
  const { user } = UserAuth() ?? {};

  useEffect(() => {
    setTitle(media.title || media.name || media.mediaTitle);

    setPosterPath(
      apiConfig.w500Image(
        media.poster_path ||
          media.backdrop_path ||
          media.mediaPoster ||
          media.profile_path
      )
    );

    if (mediaType === "movie") {
      setReleaseDate(media.release_date && media.release_date.split("-")[0]);
    } else {
      setReleaseDate(
        media.first_air_date && media.first_air_date.split("-")[0]
      );
    }
  }, [media, mediaType]);

  return (
    <div
      className="movie-card"
      style={{
        backgroundImage:
          media.poster_path ||
          media.backdrop_path ||
          media.mediaPoster ||
          media.profile_path
            ? `url(${posterPath})`
            : `url(${photoComingSoon})`,
        marginBottom: "2rem",
        height: "25rem",
        textShadow: "10px",
      }}
    >
      {/* movie or tv item */}
      {media && (
        <>
          <div>
            <Link
              to={mediaType !== "person" ? "/detail" : "/personDetails"}
              state={{
                item: mediaType !== "person" ? media : media.id,
                MediaType: mediaType,
              }}
            >
              {user ? (
                <Button>
                  <Typography
                    variant="h6"
                    style={{ fontWeight: "bold", marginTop: "1rem" }}
                  >
                    Read More
                  </Typography>
                  <div style={{ paddingTop: "3px" }}>
                    <ReadMoreIcon fontSize="large" />
                  </div>
                </Button>
              ) : (
                <OutlineButton className="small">
                  <div>
                    <Link to="/login">sign in to view more</Link>
                  </div>
                </OutlineButton>
              )}
            </Link>
            <h2>{releaseDate ? "(" + releaseDate + ")" : ""}</h2>
          </div>
          <h3>{title}</h3>
        </>
      )}
      {/* movie or tv item */}
    </div>
  );
};

export default MediaItem;
