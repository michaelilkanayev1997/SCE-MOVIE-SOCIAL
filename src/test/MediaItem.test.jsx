import { configure, shallow } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import React from "react";
import MediaItem from "../components/MediaItem";
import "jest-extended";
import { Link } from "react-router-dom";

configure({ adapter: new Adapter() });

describe("MediaItem component", () => {
    it('should render without errors', () => {
        const media = { title: 'Movie title', poster_path: '/path/to/poster.jpg' };
        const mediaType = 'movie';
        const wrapper = shallow(<MediaItem media={media} mediaType={mediaType} />);
        expect(wrapper).toHaveLength(1);
    });

    it('should render a link with the correct props', () => {
        const media = { id: 123 };
        const mediaType = 'movie';
        const wrapper = shallow(<MediaItem media={media} mediaType={mediaType} />);
        expect(wrapper.find(Link)).toHaveLength(2);
        
      });

      const movieMedia = {
        title: 'The Matrix',
        poster_path: '/f89U3ADr1oiB1s9GkdPOEpXUk5H.jpg',
        release_date: '1999-03-30',
      };
      const tvMedia = {
        name: 'Stranger Things',
        backdrop_path: '/9Wg5CRFixInsZBYKSEiUDwqvIrx.jpg',
        first_air_date: '2016-07-15',
      };
      
      it('should render correctly with movie media', () => {
        const wrapper = shallow(<MediaItem media={movieMedia} mediaType="movie" />);
        expect(wrapper).toMatchSnapshot();
      });
      
      it('should render correctly with tv media', () => {
        const wrapper = shallow(<MediaItem media={tvMedia} mediaType="tv" />);
        expect(wrapper).toMatchSnapshot();
      });
      
      it('should display the "photo-coming-soon" image when poster path is not provided', () => {
        const mediaWithoutPoster = { title: 'Movie without poster' };
        const wrapper = shallow(<MediaItem media={mediaWithoutPoster} mediaType="movie" />);
        expect(wrapper.find('.movie-card').prop('style').backgroundImage).toContain('photo-coming-soon.jpg');
      });
      
      it("sets the poster path correctly", () => {
        const media = {
          poster_path: "poster_path.jpg",
          backdrop_path: "",
          mediaPoster: "",
          profile_path: "",
        };
        const wrapper = shallow(<MediaItem media={media} />);
        const posterPath = wrapper.find(".movie-card").prop("style").backgroundImage;
    
        });

});