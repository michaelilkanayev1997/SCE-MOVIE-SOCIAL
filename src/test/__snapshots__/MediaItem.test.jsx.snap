// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaItem component should render correctly with movie media 1`] = `ShallowWrapper {}`;

exports[`MediaItem component should render correctly with tv media 1`] = `ShallowWrapper {}`;
